.github/actions/setup-deps/action.yml

```diff
@@ -0,0 +1,64 @@
+name: "Turborepo CI Setup"
+description: "Sets up everything for CI"
+inputs:
+  github-token:
+    description: "Github token. You can pass secrets.GITHUB_TOKEN"
+    required: true
+  setup-node:
+    description: "Set to false to prevent Node setup steps"
+    required: false
+    default: true
```

Comment:
How about we split this into `setup-go` and `setup-node` (and eventually, `setup-rust`, `setup-python` etc.)

Reply:
Example usage:
```
      - uses: ./.github/actions/setup-go
        with:
          github-token: ${{ secrets.GITHUB_TOKEN }}
      - uses: ./.github/actions/setup-node
      - uses: ./.github/actions/setup-rust

```